swank.asd

```diff
@@ -19,19 +19,87 @@
 ;; This code has been placed in the Public Domain.  All warranties
 ;; are disclaimed.
 
-(defpackage :swank-loader
-  (:use :cl))
+(defpackage #:swank-loader
+  (:use #:common-lisp #:asdf)
+  (:export #:init #:*source-directory*))
 
-(in-package :swank-loader)
+(in-package #:swank-loader)
 
-(defclass swank-loader-file (asdf:cl-source-file) ())
+(defun load-user-init-file ()
+  "Load the user init file, return NIL if it does not exist."
+  (load (merge-pathnames (user-homedir-pathname)
+                         (make-pathname :name ".swank" :type "lisp"))
+        :if-does-not-exist nil))
 
-;;;; after loading run init
+(defvar *source-directory*)
 
-(defmethod asdf:perform ((o asdf:load-op) (f swank-loader-file))
-  (load (asdf::component-pathname f))
-  (funcall (read-from-string "swank-loader::init") :reload t))
+(defclass no-load-file (cl-source-file) ())
```

Comment:
What's this for?

Reply:
To make ASDF compile the contribs but not load them. A change to swank.lisp adds the function find-module-asdf to load the contribs.

Reply:
Actually the use of no-load-file and listing all contribs in the system definition is unnecessary because swank-require will load the source file if there's no fasl. Compiling the contribs makes M-x slime about 1s faster.

Reply:
Ah, it makes more sense now with that `perform` method. Do you think `compile-only` would be useful? Have you considered splitting each contrib into its own system? Perhaps using ASDF's ability to define multiple systems in a single .asd?

Reply:
The use of no-load-file allows keeping the current contrib loading mechanism with minimal changes and backwards compatibility with swank-loader.lisp, for those who don't want to use ASDF.